Assets/Scripts/Loading/LoadLevel.cs

```diff
@@ -0,0 +1,52 @@
+// Morgan Houston
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+
+public class LoadLevel : MonoBehaviour
+{
+    #region Fields
+    [Tooltip("Player Sprite that follows loading bar.")]
+    public GameObject player;
+    [Tooltip("The end of the loading bar.")]
+    public GameObject loadingBarEnd;
+    [Tooltip("The loading bar.")]
+    public Slider loadingBar;
+    private float progress;
+
+    #endregion
+
+    #region Unity Methods
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        StartCoroutine(LoadAsyncScene(SceneLoader.levelToLoad));
+    }
+
+    #endregion
+
+    #region Methods
+
+    IEnumerator LoadAsyncScene(int index)
+    {
+        yield return null;
+
+        AsyncOperation loading = SceneManager.LoadSceneAsync(index);
+
+        while(!loading.isDone)
+        {
+            progress = Mathf.Clamp(loading.progress / .9f, 0f, 0.9999f);
```

Comment:
In here I have no Idea why you choose those numbers
you could have constants named ALMOST_COMPLETE, PERCENT_90, NOT_STARTED, A_BIG_NUMBER or something to help the others know why are those numbers there without having to think too much about it.

Reply:
Thank you, I will update. I used them to tell if Almost complete, sometimes it gets stuck at 100% when scenes get bigger.